Rx/v2/src/rxcpp/operators/rx-sequence_equal.hpp

```diff
@@ -0,0 +1,215 @@
+struct sequence_equal : public operator_base<bool>
+{
+    typedef rxu::decay_t<Observable> source_type;
+    typedef rxu::decay_t<T> source_value_type;
+    typedef rxu::decay_t<OtherObservable> other_source_type;
+    typedef typename other_source_type::value_type other_source_value_type;
+    typedef rxu::decay_t<Coordination> coordination_type;
+    typedef typename coordination_type::coordinator_type coordinator_type;
+
+    struct values {
+        values(source_type s, other_source_type t, coordination_type sf)
+                : source(std::move(s))
+                , other(std::move(t))
+                , coordination(std::move(sf))
+        {
+        }
+
+        source_type source;
+        other_source_type other;
+        coordination_type coordination;
+    };
+
+    values initial;
+
+    sequence_equal(source_type s, other_source_type t, coordination_type sf)
+        : initial(std::move(s), std::move(t), std::move(sf))
+    {
+    }
+
+    template<class Subscriber>
+    void on_subscribe(Subscriber s) const {
+
+        typedef Subscriber output_type;
+
+        struct state_type
+            : public std::enable_shared_from_this<state_type>
+            , public values
+        {
+            state_type(const values& vals, coordinator_type coor, const output_type& o)
+                : values(vals)
+                , coordinator(std::move(coor))
+                , out(o)
+                , source_completed(false)
+                , other_completed(false)
+            {
+                out.add(other_lifetime);
+                out.add(source_lifetime);
+            }
+
+            composite_subscription other_lifetime;
+            composite_subscription source_lifetime;
+            coordinator_type coordinator;
+            output_type out;
+            
+            mutable std::list<source_value_type> source_values;
+            mutable std::list<other_source_value_type> other_values;
+            mutable bool source_completed;
+            mutable bool other_completed;
+        };
+
+        auto coordinator = initial.coordination.create_coordinator();
+        auto state = std::make_shared<state_type>(initial, std::move(coordinator), std::move(s));
+
+        auto other = on_exception(
+            [&](){ return state->coordinator.in(state->other); },
+            state->out);
+        if (other.empty()) {
+            return;
+        }
+
+        auto source = on_exception(
+            [&](){ return state->coordinator.in(state->source); },
+            state->out);
+        if (source.empty()) {
+            return;
+        }
+
+        auto check_equal = [state]() {
+            if(!state->source_values.empty() && !state->other_values.empty()) {
+                auto x = std::move(state->source_values.front());
+                state->source_values.pop_front();
+
+                auto y = std::move(state->other_values.front());
+                state->other_values.pop_front();
+
+                if (x != y) {
```

Comment:
when you change to the binary predicate please change this to !pred(x, y) or if this line of code remains, change to !(x == y)

In either case I would like == to be the default operator used instead of !=